Fix Crash when a Customization TextField is empty

Shad0wlife opened the pull request:
The empty string caused the  String parsing for the textfields
- expectedBaseHeight
- expectedHeightVariation
- actualHeight

to fail with a NumberFormatException, resulting in a crash.
This fix catches that exception, turns the cursor in the field red (and as soon as a proper value is detected back to grey) and if the menu is left (by clicking Done) with an empty field, that field resets to default.

Shad0wlife committed:
Fix Crash when a Customization TextField is empty

The empty string caused the  String parsing for the textfields
- expectedBaseHeight
- expectedHeightVariation
- actualHeight
to fail with a NumberFormatException, resulting in a crash.
This fix catches that exception, turns the cursor in the field red (and as soon as a proper value is detected back to grey) and if the menu is left with an empty field, that field resets to default.

Niko-sk2x commented:
I really don't like the idea of catching an exception as a normal expected behavior. But can't fix stupid Java library design here...

Niko-sk2x commented on src/main/java/io/github/opencubicchunks/cubicchunks/cubicgen/customcubic/gui/AdvancedTerrainShapeTab.java:
```diff
@@ -461,9 +461,31 @@ private void setLockedXZ(boolean lock) {
+			this.actualHeight.setCursorColor(0xD0D0D0);
+		}catch(NumberFormatException e){
+			this.actualHeight.setCursorColor(0xD00000);
+		}
```
Formatting

Shad0wlife commented:
I actually don't like the catching as a solution either, but any alternative would really only result in wrapping the catching in some method - as you said, sadly that behaviour is Java sided.
If you would prefer that over the try/catch - the alternative would be getting the string first and then checking if it's empty and acting based on that. Would basically be the same thing - just not with an exception.

```
String textValue = this.textfield.getText();
if(textValue.equals("")){
    //Red cursor
}else{
    //Save value and normal cursor
}
```

Also, what do you think about the red cursor? Ok like that or too much?

Shad0wlife committed:
Formatting

Formatting based on other CC code.
I did this quickly on my notebook with Npp, I don't have idea installed and I don't want to try setting up eclipse for this, so no automatic code formatter - if there's any other formatting thing, please tell me so.

Niko-sk2x commented:
Red cursor is fine. As for is empty check - that would also break. With too big or too small integers.

Niko-sk2x commented:
It still has tabs

Shad0wlife committed:
Formatting: Tabs -> 4 Spaces

Niko-sk2x commented:
Did you actually test it? I don't see anything that handles doing things while typing into the text fields and I'm pretty sure there was something.

Shad0wlife commented:
How do you mean "handles doing things while typing into the text field"?
The modified method seems to be the only one to actually parse the field contents.
If you mean by testing that I 

- ran it
- emptied the fields, the cursor turned red - and grey again when I entered something
- and that after leaving the menu, it's back at default

then yes.
If you want an actual testing procedure, please give me guidelines for that - just be aware that I don't know how to actually write unit tests :)

Niko-sk2x commented:
Unit tests in Minecraft mods are rare so I don't demand it. If it works, it's ok.
